src/EditorFeatures/Core/EditorConfigSettings/DataProvider/Analyzer/AnalyzerSettingsProvider.cs

```diff
@@ -60,7 +60,7 @@ private IEnumerable<AnalyzerSetting> GetSettings(AnalyzerReference analyzerRefer
                         var selectedDiagnostic = g.First();
                         var isEditorconfig = selectedDiagnostic.IsDefinedInEditorConfig(editorConfigOptions);
                         var settingLocation = new SettingLocation(isEditorconfig ? LocationKind.EditorConfig : LocationKind.VisualStudio, FileName);
-                        var severity = selectedDiagnostic.GetEffectiveSeverity(editorConfigOptions);
+                        var severity = selectedDiagnostic.GetEffectiveSeverity(editorConfigOptions, providerAndTree: null);
```

Comment:
💭 I'm not sure I understand the impact of passing `null` here

Reply:
I can add a doc comment and make this an optional parameter too to clarify this. If the provider and tree are provided, the `dotnet_diagnostic` based entries are looked up from the syntax tree provider for the given tree, otherwise they are looked up from the `editorConfigOptions` which just contains a string to string dictionary of options. This is due to the way compiler parses the editorconfig file and separates out the `dotnet_diagnostic` entries into a separate dictionary from ruleID string to ReportDiagnostic, while maintains a string to string dictionary for rest of the options. `AnalyzerSettingsProvider` type in question here is operating on a constructed editorConfigOptions not specific to a particular tree, so the providerAndTree doesn't apply to it.